docs/proposals/componentconfig-dump.md

```diff
@@ -0,0 +1,101 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should refer to the docs that go with that version.
+
+Documentation for other releases can be found at [releases.K8s.io](http://releases.K8s.io).
+</strong>
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+
+# Overview
+
+We describe a proposal for retrieving component configurations (componentconfigs) that govern the behavior of any one of the Kubernetes (K8s) components. 
+
+We propose several important changes:
+* adding new authenticated protected path to a component api: 
+    * `/componentconfig` 
+    * Discussions to be held with SIG-Auth individuals about making such a path secured, but we will invoke that similar discussions around metrics have taken place.
+* helping complete the migration of flags to componentconfigs
+
+## Background
+
+The document is built from SIG-Scale and PR discussions. Here we document the results of these and build the proposal of what changes should/will be made to K8s.
+
+## Motivation
+
+By necessity Kubernetes is a stateful thing made up of components who have been configured in a particular way. Currently the K8s code base, as 09/2016, does not support a way nor means by which to query for the set of configurations that led to the current state of any component. As a consequence it is extremely difficult to debug the current state of a K8s cluster, especially when original configurations have been lost, or are simply not available. 
+
+## Design
+
+Components will have either a path indicated by annotations that can point to how to retrieve the componentconfigs or through a well known component api path, as discussed below.
+
+### Component API changes
+
+Components (like kubelet) have endpoints that can be altered to include an endpoint such as
+
+```
+ /componentconfigs 
+```
+
+similar to /metrics endpoints for K8s components and likely to be authenticated in much the same way.
```

Comment:
Unlike metrics, we're planning to explicitly version component configuration, and we're planning to break up the monolithic configuration types that were just transliterated from existing flags into semantically grouped types. 

Have you thought about:
a) How a client would request a specific version of the configuration?
b) How multiple config objects would be represented and requested?